Library/Homebrew/cmd/list.rb

```diff
@@ -35,18 +50,18 @@ def list
       return
     end
 
-    if ARGV.include?("--pinned") || ARGV.include?("--versions")
-      filtered_list
+    if args.pinned? || args.versions?
+      filtered_list(args)
```

Comment:
Could make `args` -> `@args`, instance variable, then its not needed to pass it as an argument.